data/transactions/logic/eval_test.go

```diff
@@ -5117,3 +5119,267 @@ func TestOpBase64Decode(t *testing.T) {
+	sb := strings.Builder{}
+	ep := defaultEvalParams(&sb, &txn)
+	ep.Proto = &proto
+	ep.Ledger = ledger
+	testCases := []struct {
+		source string
+	}{
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"algo\",\"key2\":{\"key3\": \"teal\", \"key4\":3}, \"key5\": 18446744073709551615 }"; 
+			byte "key0"; 
+			json_ref JSONUint64; 
+			int 0; 
+			==`,
+		},
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"algo\",\"key2\":{\"key3\": \"teal\", \"key4\": 3}, \"key5\": 18446744073709551615 }"; 
+			byte "key5"; 
+			json_ref JSONUint64; 
+			int 18446744073709551615; //max uint64 value
+			==`,
+		},
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"algo\",\"key2\":{\"key3\": \"teal\", \"key4\": 3}, \"key5\": 18446744073709551615 }"; 
+			byte "key1"; 
+			json_ref JSONString; 
+			byte "algo"; 
+			==`,
+		},
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"\\u0061\\u006C\\u0067\\u006F\",\"key2\":{\"key3\": \"teal\", \"key4\": 3}, \"key5\": 18446744073709551615 }"; 
+			byte "key1"; 
+			json_ref JSONString; 
+			byte "algo"; 
+			==`,
+		},
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"algo\",\"key2\":{\"key3\": \"teal\", \"key4\": {\"key40\": 10}}, \"key5\": 18446744073709551615 }"; 
+			byte "key2"; 
+			json_ref JSONObject; 
+			byte "key4";
+			json_ref JSONObject;
+			byte "key40";
+			json_ref JSONUint64
+			int 10
+			==`,
+		},
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"algo\",\"key2\":{\"key3\": \"teal\", \"key4\": {\"key40\": 10}}, \"key5\": 18446744073709551615 }"; 
+			byte "key2"; 
+			json_ref JSONObject; 
+			byte "key3";
+			json_ref JSONString;
+			byte "teal"
+			==`,
+		},
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"algo\",\"key2\":{\"key3\": \"\\"teal\\"\", \"key4\": {\"key40\": 10}}, \"key5\": 18446744073709551615 }"; 
+			byte "key2"; 
+			json_ref JSONObject; 
+			byte "key3";
+			json_ref JSONString;
+			byte ""teal"" // quotes match
+			==`,
+		},
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"algo\",\"key2\":{\"key3\": \" teal \", \"key4\": {\"key40\": 10}}, \"key5\": 18446744073709551615 }"; 
+			byte "key2"; 
+			json_ref JSONObject; 
+			byte "key3";
+			json_ref JSONString;
+			byte " teal " // spaces match
+			==`,
+		},
+		{
+			source: `byte  "{\"key0\": 0,\"key1\": \"algo\",\"key2\":{\"key3\": \"teal\", \"key4\": {\"key40\": 10, \"key40\": \"10\"}}, \"key5\": 18446744073709551615 }"; 
+			byte "key2"; 
+			json_ref JSONObject; 
+			byte "key4";
+			json_ref JSONObject;
+			byte "{\"key40\": 10, \"key40\": \"10\"}"
+			==
+			`,
+		},
+		{
+			source: `byte  "{\"rawId\": \"responseId\",\"id\": \"0\",\"response\": {\"attestationObject\": \"based64url_encoded_buffer\",\"clientDataJSON\":  \" based64url_encoded_client_data\"},\"getClientExtensionResults\": {},\"type\": \"public-key\"}"; 
+			byte "response"; 
+			json_ref JSONObject; 
+			byte "{\"attestationObject\": \"based64url_encoded_buffer\",\"clientDataJSON\":  \" based64url_encoded_client_data\"}" // object as it appeared in input
```

Comment:
I'd also suggest testing that a unicode escape sequence remains in the JSON string. E.g. use `\u0061` instead of `a` somewhere. I used this site to find that: https://dencode.com/en/string/unicode-escape

You could also check that unicode escape sequences in the keys get handled properly, like if `\u0072\u0065\u0073\u0070\u006f\u006e\u0073\u0065` was used instead of `response` in the JSON source.